bind/mount.go

```diff
@@ -36,11 +36,6 @@ func SetupIntermediateMountNamespace(spec *specs.Spec, bundlePath string) (unmou
 		return nil, errors.Wrapf(err, "error creating new mount namespace for %v", spec.Process.Args)
 	}
 
-	// Make all of our mounts private to our namespace.
-	if err := mount.MakeRPrivate("/"); err != nil {
```

Comment:
Should this be a MakeRSlave?

We don't want mounts inside of the container to leak to the host.  If the host is SHARED, won't they leak?

Reply:
if we make it rslave, the mounts will still be unaccessible from the outside.

This is needed to propagate the rootfs and I think it is equivalent to what we do in Podman/CRI-O where the .  The OCI runtime will still set the root mount to rslave/private.

Leaking mounts though is a possibility, let me see if I can address it in a different way..